File: 백준/Bronze/6131. 완전 제곱수/완전 제곱수.c
#include <stdio.h>
int main(void) {
    int n, a = 1, b = 2, ans = 0;
    scanf("%d", &n);
    while (a != b) {
        if (b * b - a * a == n) {
            ans += 1;
            b += 1;
        } else if (b * b - a * a > n) {
            a += 1;
        } else {
            b += 1;
        }
    }
    printf("%d", ans);
    return 0;
}

File: 백준/Bronze/6131. 완전 제곱수/완전 제곱수.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n=int(input())
a,b,ans=1,2,0
while a!=b:
    if b**2-a**2==n:
        ans+=1
        b+=1
    elif b**2-a**2>n: a+=1
    else: b+=1
print(ans)